### PR TITLE
add support to get multiple workflow instances by a list of ids

### PR DESCRIPTION
**Pull request by knoxi**
Added the possibility to get multiple workflow instances by a list of ids ( issue #261 )

### CODE REVIEW

File: src/WorkflowCore/Services/DefaultProviders/MemoryPersistenceProvider.cs [hunk 1]
```diff
@@ -61,6 +61,19 @@ public async Task<WorkflowInstance> GetWorkflowInstance(string Id)
...
+        public async Task<IEnumerable<WorkflowInstance>> GetWorkflowInstances(IEnumerable<string> ids)
+        {
+            if (ids == null)
+            {
+                return new List<WorkflowInstance>();
+            }
+
+            lock (_instances)
+            {
+                return _instances.Where(x => ids.Contains(x.Id, StringComparer.OrdinalIgnoreCase));
```
Comment (danielgerlag) [inline src/WorkflowCore/Services/DefaultProviders/MemoryPersistenceProvider.cs]: why would we ignore case?

File: src/providers/WorkflowCore.Persistence.MongoDB/Services/MongoPersistenceProvider.cs [hunk 2]
```diff
@@ -113,7 +113,18 @@ public async Task<WorkflowInstance> GetWorkflowInstance(string Id)
...
-        
+
+        public async Task<IEnumerable<WorkflowInstance>> GetWorkflowInstances(IEnumerable<string> ids)
+        {
+            if (ids == null)
+            {
+                return new List<WorkflowInstance>();
+            }
+
+            var result = await WorkflowInstances.FindAsync(x => ids.Contains(x.Id, StringComparer.OrdinalIgnoreCase));
```
Comment (danielgerlag) [inline src/providers/WorkflowCore.Persistence.MongoDB/Services/MongoPersistenceProvider.cs]: why ignore case?

**Comment by knoxi** [comment]
I extended the testing in `WorkflowCore.UnitTests.BasePersistenceFixture` for the new `GetWorkflowInstances`
Also I removed ignore case in the mentioned provieders.

File: src/providers/WorkflowCore.Persistence.MongoDB/Services/MongoPersistenceProvider.cs (hunk 2 above)
Comment (knoxi) [reply src/providers/WorkflowCore.Persistence.MongoDB/Services/MongoPersistenceProvider.cs]: removed

File: src/WorkflowCore/Services/DefaultProviders/MemoryPersistenceProvider.cs (hunk 1 above)
Comment (knoxi) [reply src/WorkflowCore/Services/DefaultProviders/MemoryPersistenceProvider.cs]: removed

**Comment by danielgerlag** [comment]
Tests against DynamoDb are failing

**Comment by knoxi** [comment]
After dealing a long time to connect to the DynamoDB container, I fixed now the DynamoDB provider

File: src/WorkflowCore/Interface/IPersistenceProvider.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@ namespace WorkflowCore.Interface
 {
     /// <remarks>
     /// The implemention of this interface will be responsible for
-    /// persisiting running workflow instances to a durable store    
+    /// persisiting running workflow instances to a durable store
```
Comment (germandb) [inline src/WorkflowCore/Interface/IPersistenceProvider.cs]: persisting

Comment (knoxi) [reply src/WorkflowCore/Interface/IPersistenceProvider.cs]: fix typo

**Pull request merged**